.github/workflows/iroha2-pr.yml

```diff
@@ -8,46 +8,131 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  long-tests:
-    runs-on: ubuntu-latest
-    container: rust:1.56-buster
+  cli-test:
+    runs-on: [self-hosted, linux]
+    container:
+      image: 7272721/i2-ci:latest
     steps:
       - uses: actions/checkout@v2
-      - name: Install dependencies
+      - uses: Swatinem/rust-cache@v1
+      - name: Build Client CLI
+        run: mold -run cargo build
+        working-directory: client_cli
+      - name: Build Iroha Peer
+        run: mold -run cargo build
+        working-directory: cli
+      - name: Mark both binaries as executable
         run: |
-          apt-get update
-          apt-get install -y --no-install-recommends \
-            build-essential \
-            ca-certificates \
-            clang \
-            llvm-dev
-      - name: Run long tests
-        run: cargo test --workspace --verbose --no-fail-fast -- --ignored --test-threads=1 long
+          chmod +x target/debug/iroha
+          chmod +x target/debug/iroha_client_cli
+      - name: Setup test environment
+        run: bash -c './scripts/test_env.sh setup'
+      - name: Genesis test
+        run: bash -c './scripts/tests/genesis.sh || (cat test/peers/iroha0.log; false )'
+      - name: Basic register and mint
+        run: bash -c './scripts/tests/register_mint_quantity.sh || (cat test/peers/iroha0.log; false )'
+        if: always()
+      - name: Cleanup test environment
+        run: bash -c './scripts/test_env.sh cleanup'
```

Comment:
always?

Reply:
Since this doesn't cache on fail, it would be redundant.